### ISSUE TITLE
Recompute metrics each time the analytics tab is loaded

### ISSUE THREAD

**Issue by eljhkrr** [medic/cht-core#6683]
**What feature do you want to improve?**
A scenario has been encountered where a user views in-app analytics, performs an action and then returns to the in-app analytics tab expecting that the action performed would lead to an update of analytics displayed as detailed in https://github.com/medic/config-pih/issues/163. The user views stale metrics and is confused whether the action performed was registered or in-app analytics are working well.

**Describe the improvement you'd like**
A user should be able to refresh metrics displayed via a button or pull to refresh.

**Describe alternatives you've considered**
A timestamp showing when the metrics were computed would also be useful in giving the user feedback.

**Comment by garethbowen** [comment]
This is a bug - targets should be refreshed every time the page is loaded and should not require any refresh button or "last updated" timestamp.

@eljhkrr Can you please provide full reproduction steps, screenshots, and logs?

**Comment by dianabarsan** [comment]
@garethbowen @eljhkrr 

I added a comment into the linked issue (https://github.com/medic/config-pih/issues/163#issuecomment-712650843): 

From the issue, it appears that there is a target that relies on counting muted contacts. 

> Muting a synchronous action that only happens on the server. So the client (the phone) needs to sync the muting report up, the report needs to be picked up by Sentinel and then the client needs to sync the report down in order for the Targets page to show accurate numbers.

**Comment by eljhkrr** [comment]
Yeah, this was related to muting a household and observing that household counts were not updating. For future reference I'm listing exact steps to reproduce here

1. Open analytics tab. Check household count with the following config in `targets.js` uploaded
```
{
  id: 'households-registered-all-time',
  translation_key: 'targets.household.registrations.title',
  subtitle_translation_key: 'targets.all_time.subtitle',
  type: 'count',
  icon: 'medic-clinic',
  goal: -1,
  appliesTo: 'contacts',
  appliesToType: ['household'],
  appliesIf: function(c){
    return !isMuted(c.contact);
  },
  date: 'now'
},
```
2. Switch to the contacts tab and mute a number of households.
3. Tap on analytics tab, the number of households should be reduced by the number of households muted in step 2. However, the count of households reported remains the same as in step 1.

Ideally, metrics should be updated without requiring reports to be synced.

**Comment by dianabarsan** [comment]
If indeed it's related to muting, it's not a bug, but an improvement as you tagged it before. 
However, I'm not sure how muting would work without syncing.

**Comment by kennsippell** [comment]
Should this be considered a duplicate of https://github.com/medic/cht-core/issues/6737? Sounds like the same underlying cause. #6937 seems to articulates the issue clearly; while this seems to be an excellent example of its impacts.

**Comment by abbyad** [comment]
Yes, closing as this is an example situation that would be solved by #6737, which is essentially a request for muting to be handled in some way before syncing.
